pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -109,7 +109,7 @@ public void init(HelixManager helixManager) {
   @Override
   public synchronized void processClusterChange(HelixConstants.ChangeType changeType) {
     Preconditions.checkState(changeType == HelixConstants.ChangeType.EXTERNAL_VIEW
-        || changeType == HelixConstants.ChangeType.INSTANCE_CONFIG, "Illegal change type: " + changeType);
+        || changeType == HelixConstants.ChangeType.INSTANCE_CONFIG, "Illegal change type: %s", changeType);
```

Comment:
Cool, i didn't know that `Precondition` supports the formatting.

Reply:
It only supports `%s`